src/main/java/redis/clients/jedis/JedisPubSubBase.java

```diff
@@ -69,11 +76,23 @@ public final void punsubscribe(T... patterns) {
   }
 
   public final void ping() {
-    sendAndFlushCommand(Command.PING);
+    authenticator.commandSync.lock();
```

Comment:
IMHO it would be good to mention why we need to lock here.

As far as I can tell this is because the AUTH command is async and we want to be sure we dont get a concurrency problem when the user and auth threads attempt to write in the same time?

Reply:
thats correct , i will provide explanation in another iteration